src/main/java/org/folio/consortia/utils/TenantContextUtils.java

```diff
@@ -0,0 +1,52 @@
+package org.folio.consortia.utils;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.folio.spring.DefaultFolioExecutionContext;
+import org.folio.spring.FolioExecutionContext;
+import org.folio.spring.FolioModuleMetadata;
+import org.folio.spring.integration.XOkapiHeaders;
+import org.folio.spring.scope.FolioExecutionContextSetter;
+import org.springframework.messaging.MessageHeaders;
+
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class TenantContextUtils {
+
+  public static FolioExecutionContext getFolioExecutionContextCreatePrimaryAffiliationEvent(
+                                                                                  MessageHeaders headers,
+                                                                                  FolioModuleMetadata moduleMetadata) {
+    return getContextFromKafkaHeaders(headers, moduleMetadata);
+  }
+
+  public static void runInFolioContext(FolioExecutionContext context, Runnable runnable) {
+    try (var fec = new FolioExecutionContextSetter(context)) {
```

Comment:
This context is set in the current thread and then removed after execution, so looks like current thread won't have any context set after this method. Is this expected?